### PR TITLE
Fixed TypeError in draw_value

### LINKED ISSUE #2151: Drawing binomial samples from dist raises TypeError
The `random` method for `Binomial` is broken (at least when used as a `dist` object). Using `len` to query the size of the parameter does not work when it is a tensor.

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-2c304bd573b6> in <module>()
----> 1 x.random()

/Users/fonnescj/anaconda3/envs/stat_pycon/lib/python3.6/site-packages/pymc3/distributions/discrete.py in random(self, point, size, repeat)
     46 
     47     def random(self, point=None, size=None, repeat=None):
---> 48         n, p = draw_values([self.n, self.p], point=point)
     49         return generate_samples(stats.binom.rvs, n=n, p=p,
     50                                 dist_shape=self.shape,

/Users/fonnescj/anaconda3/envs/stat_pycon/lib/python3.6/site-packages/pymc3/distributions/distribution.py in draw_values(params, point)
    188         # "Homogonise" output
    189         values[i] = np.atleast_1d(draw_value(
--> 190             param, point=point, givens=givens.values()))
    191     if len(values) == 1:
    192         return values[0]

/Users/fonnescj/anaconda3/envs/stat_pycon/lib/python3.6/site-packages/pymc3/distributions/distribution.py in draw_value(param, point, givens)
    251         except:
    252             shape = param.shape
--> 253         if len(shape) == 0 and len(value) == 1:
    254             value = value[0]
    255     return value

TypeError: object of type 'TensorVariable' has no len()
```

Running master on macOS 10.12.4 and Python 3.6.

### PR DESCRIPTION
**Pull request by fonnesbeck**
Closes #2151

### CODE REVIEW

File: pymc3/distributions/distribution.py [hunk 1]
```diff
@@ -248,8 +248,11 @@ def draw_value(param, point=None, givens=()):
     if hasattr(param, 'shape'):
         try:
             shape = param.shape.tag.test_value
-        except:
-            shape = param.shape
+        except AttributeError:
+            try:
+                shape = param.shape.eval()
```
Comment (ferrine) [inline pymc3/distributions/distribution.py:253]: What about missing input exception here?

Comment (fonnesbeck) [reply pymc3/distributions/distribution.py:253]: Which input exception are you referring to?

**Comment by fonnesbeck** [comment]
Can this be merged? The `random` method remains broken otherwise.

**Comment by twiecki** [comment]
LGTM.

**Pull request merged**